pkg/cmd/get/secrets.go

```diff
@@ -0,0 +1,198 @@
+	Short: "retrieve secrets from the cluster",
+	Long:  ``,
+	RunE:  getSecretsE,
+}
+
+// well known secrets that are part of the core packages
+var corePkgSecrets = map[string][]string{
+	"argocd": []string{"argocd-initial-admin-secret"},
+	"gitea":  []string{"gitea-admin-secret"},
+}
+
+type TemplateData struct {
+	Name      string
+	Namespace string
+	Data      map[string]string
+}
+
+func getSecretsE(cmd *cobra.Command, args []string) error {
+	ctx, ctxCancel := context.WithCancel(ctrl.SetupSignalHandler())
+	defer ctxCancel()
+	kubeConfigPath := filepath.Join(homedir.HomeDir(), ".kube", "config")
+
+	b := build.NewBuild("", "", kubeConfigPath, "", "", util.CorePackageTemplateConfig{}, []string{}, false, k8s.GetScheme(), ctxCancel)
+
+	kubeConfig, err := b.GetKubeConfig()
+	if err != nil {
+		return fmt.Errorf("getting kube config: %w", err)
+	}
+
+	kubeClient, err := b.GetKubeClient(kubeConfig)
+	if err != nil {
+		return fmt.Errorf("getting kube client: %w", err)
+	}
+
+	if len(packages) == 0 {
+		return printAllPackageSecrets(ctx, os.Stdout, kubeClient)
+	}
+
+	return printPackageSecrets(ctx, os.Stdout, kubeClient)
+}
+
+func printAllPackageSecrets(ctx context.Context, outWriter io.Writer, kubeClient client.Client) error {
+	selector := labels.NewSelector()
+	secretsToPrint := make([]any, 0, 2)
+
+	for k, v := range corePkgSecrets {
+		for i := range v {
+			secret, sErr := getSecretByName(ctx, kubeClient, k, v[i])
+			if sErr != nil {
+				if errors.IsNotFound(sErr) {
+					continue
+				}
+				return fmt.Errorf("getting secret %s in %s: %w", v[i], k, sErr)
+			}
+			secretsToPrint = append(secretsToPrint, secretToTemplateData(secret))
+		}
+	}
+
+	secrets, err := getSecretsByCNOELabel(ctx, kubeClient, selector)
+	if err != nil {
+		return fmt.Errorf("listing secrets: %w", err)
+	}
+
+	for i := range secrets.Items {
+		secretsToPrint = append(secretsToPrint, secretToTemplateData(secrets.Items[i]))
+	}
+
+	if len(secretsToPrint) == 0 {
+		fmt.Println("no secrets found")
+		return nil
+	}
+	return renderTemplate(secretTemplatePath, outWriter, secretsToPrint)
+}
+
+func printPackageSecrets(ctx context.Context, outWriter io.Writer, kubeClient client.Client) error {
+	selector := labels.NewSelector()
+	secretsToPrint := make([]any, 0, 2)
+
+	for i := range packages {
+		p := packages[i]
+		secretNames, ok := corePkgSecrets[p]
+		if ok {
+			for j := range secretNames {
+				secret, sErr := getSecretByName(ctx, kubeClient, p, secretNames[j])
+				if sErr != nil {
+					if errors.IsNotFound(sErr) {
+						continue
+					}
+					return fmt.Errorf("getting secret %s in %s: %w", secretNames[j], p, sErr)
+				}
+				secretsToPrint = append(secretsToPrint, secretToTemplateData(secret))
+			}
+			continue
+		}
+
+		req, pErr := labels.NewRequirement(v1alpha1.PackageNameLabelKey, selection.Equals, []string{p})
+		if pErr != nil {
+			return fmt.Errorf("building requirement for %s: %w", p, pErr)
+		}
+
+		pkgSelector := selector.Add(*req)
+
```

Comment:
Not sure why this bit needs to sit outside `getSecretsByCNOELabel`. I'd say implement this all in that same function.